### DISCUSSION TITLE
Welcome to BioSimulations discussions!

### DISCUSSION THREAD

**Discussion by bilalshaikh42** [biosimulations/biosimulations#1743] [General]
## Welcome!
Discussions is a place for members of our community to connect with the developers and with each other. We invite you to use discussions to ask questions, share ideas, and otherwise engage with other members of the community.

To get started, we invite you to introduce yourself and your interest in BioSimulations by posting a comment below.

**Comment by jmdetloff** [comment]
Hello! My name is John. I work primarily in iOS dev, currently at Facebook, but I enjoy branching out and have experience building products with many stacks. I did a bit of lab work in university before ultimately entering a career in software development, and since then have been curious about the intersections between cs, open source, and academic research. I recently decided to pursue that interest and whole cell modeling is a domain that caught my interest in particular. I'm still quite new to the project, but Dr. Karr was kind enough to give me a rundown of the problem space, the goals of the project, and the architecture of the BioSimulations apps. I feel I've learned a ton already and am excited to continue to learn and contribute!

**Reply by bilalshaikh42** [reply]
Welcome! Excited to have you working with us. Feel free to ask any questions and peruse the open issues to see if there is anything that catches your eye to work on. We are still in the early prototyping phase, so things are pretty fluid, though they are now starting to stabilize a bit. 

We have started working on developer documentation at https://docs.biosimulations.org/developers though it is far from complete. Your perspective as a newcomer to the codebase would be very helpful in identifying the highest priority parts that need documentation,  so please let us know about anything that you find confusing/hard to get working, even if you end up figuring it out! 

Anything in particular that you are thinking of working on yet?

**Reply by jmdetloff** [reply]
Thanks! I'll put some thought into that. Off the top of my head, I think a step by step setup guide to get a user from 0 to working on one of the apps would make things more accessible. I know that's a big guide, especially when considering different configurations and starting points, but getting potential contributors building quick will likely help hook them. Another thing I got hung up on for a bit the other evening was trying to figure out how to debug through VsCode - I'll dig into this again today but if you have a resource handy or any tips that would help! 

I'm planning on digging into #3516 but I may mess around with a few other things on the way just to continue learning. For example I might tweak the drop down menus - I notice their touch target has a gap between the hover element and the drop down element, so sometimes the menu closes while moving the mouse from header to menu. Feel free to ping me on any tiny tasks you come along that you think would be good introductions to different code areas or concepts :)

**Reply by jonrkarr** [reply]
Agreed the documentation needs to be expanded. The developer documentation is thin for this repository. The existing documentation is oriented toward the collaborations we've had so far, which have focused on content (models, simulation tools, ontologies), rather than on the web application. We've organized those collaborations more de-centrally through other interfaces and repositories.

I think @bilalshaikh42 is using VsCode.

We're open to suggestions. In general, the UI could be significantly improved. We've been more focused on content (models, simulation algorithms, simulation tools, executing simulation tools, etc.) and haven't had as much bandwidth as we'd like for the UI. We anticipate that most users want to explore existing simulations (BioSimulations), followed by running simulations (runBioSimulations), and getting detailed information about simulation tools (BioSimulators). For that reason, I think improvements to BioSimulations and runBioSimulations would likely be more impactful.

**Reply by jmdetloff** [reply]
Oh whoops, I just realized the obvious reason why I cannot debug Angular code in VsCode 😅  it's running in the browser. Angular and Nest feel so similar that sometimes I forget whether I'm reading front end or back end code.

**Reply by bilalshaikh42** [reply]
> Oh whoops, I just realized the obvious reason why I cannot debug Angular code in VsCode sweat_smile it's running in the browser. Angular and Nest feel so similar that sometimes I forget whether I'm reading front end or back end code.

By Design 😛! I don't actually use VsCode's debugging features very often, but I know there are a few extensions that help on both chrome and vs code
- Chrome debugger for vscode - https://code.visualstudio.com/blogs/2016/02/23/introducing-chrome-debugger-for-vs-code
- VsCode dev tools - https://blogs.windows.com/msedgedev/2021/07/16/easier-debugging-developer-tools-in-visual-studio-code/
- Jest Runner for VSCode (helpful for running 1 test at time instead of the whole app)- https://marketplace.visualstudio.com/items?itemName=firsttris.vscode-jest-runner
- Angular Chrome extension- https://chrome.google.com/webstore/detail/angular-devtools/ienfalfjdbdpebioblfackkekamfmbnh?hl=en-US
- An alternative chrome extension - https://augury.rangle.io/
